### PR TITLE
Implemented C# version of unity_mono_method_is_inflated

### PR DESCRIPTION
**Pull request by jeanclaudegrenier**

### CODE REVIEW

File: unity/unity-embed-host/CoreCLRHost.cs [hunk 1]
```diff
@@ -671,6 +671,15 @@ public static IntPtr unity_method_get_function_pointer([NativeCallbackType("Mono
         return metBase.IsGenericMethodDefinition;
     }
 
+    [NativeFunction(nameof(unity_mono_method_is_inflated))]
+    [return: NativeCallbackType("gboolean")]
+    public static bool unity_mono_method_is_inflated(
+        [NativeCallbackType("MonoMethod*")] IntPtr method)
+    {
+        MethodBase metBase = MethodBase.GetMethodFromHandle(method.MethodHandleFromHandleIntPtr());
+        return !metBase.IsGenericMethodDefinition && metBase.IsGenericMethod;
```
Comment (scott-ferguson-unity) [inline unity/unity-embed-host/CoreCLRHost.cs]: I'm pretty sure that a non-generic method on a generic type should also return `true` for is inflated.  e.g.  `GenericAnimal<int, int>::InterfaceMethodOnIGenericAnimal` should return true.   We'll need to double check that.

**Review by joncham — APPROVED** [review]

File: unity/unity-embed-host/CoreCLRHost.cs (hunk 1 above)
Comment (jeanclaudegrenier) [reply unity/unity-embed-host/CoreCLRHost.cs]: I am getting false as the method is not generic, but am unsure if that is the expected result

Comment (jeanclaudegrenier) [reply unity/unity-embed-host/CoreCLRHost.cs]: Checked on current implementation, a non-generic method inside an inflated class is indeed considered inflated.
Will fix the PR accordingly.

File: src/coreclr/vm/mono/mono_coreclr.cpp [hunk 2]
```diff
@@ -3034,20 +3035,6 @@ extern "C" EXPORT_API gboolean EXPORT_CC unity_mono_method_is_generic(MonoMethod
     return g_HostStruct->unity_mono_method_is_generic(method);
 }
 
-extern "C" EXPORT_API gboolean EXPORT_CC unity_mono_method_is_inflated(MonoMethod* method)
```
Comment (UnityAlex) [inline src/coreclr/vm/mono/mono_coreclr.cpp:3037]: This will also need to be removed from MonoFunctions.h, mscorwks_unixexports.src and mscorwks_ntdef.src

**Review by UnityAlex — CHANGES_REQUESTED** [review]
Just need to remove references to `unity_mono_method_is_inflated` so it's no longer exported. Looks good otherwise.

**Review by UnityAlex — APPROVED** [review]

**Pull request merged**